1-js/02-first-steps/15-function-expressions/article.md

```diff
@@ -294,75 +294,75 @@ if (age < 18) {
   }
 }
 
-// Here we're out of curly braces,
-// so we can not see Function Declarations made inside of them.
+// Skończyły nam się klamerki,
+// więc nie widzimy funkcji zadeklarowanych w poprzednim bloku.
 
 *!*
 welcome(); // Error: welcome is not defined
 */!*
 ```
 
-What can we do to make `welcome` visible outside of `if`?
+Co możemy zrobić, żeby `welcome` było widoczne poza `if`em?
 
-The correct approach would be to use a Function Expression and assign `welcome` to the variable that is declared outside of `if` and has the proper visibility.
+Odpowiednim podejściem będzie użycie wyrażenia funkcyjnego i przypisanie `welcome` do zmiennej zadeklarowanej poza `if`em, która ma przez to odpowiednią widoczność.
 
-This code works as intended:
+Ten kod zadziała tak, jakbyśmy chcieli:
 
 ```js run
-let age = prompt("What is your age?", 18);
+let age = prompt("Ile masz lat?", 18);
 
 let welcome;
 
 if (age < 18) {
 
   welcome = function() {
-    alert("Hello!");
+    alert("Cześć!");
   };
 
 } else {
 
   welcome = function() {
-    alert("Greetings!");
+    alert("Pozdrowienia!");
   };
 
 }
 
 *!*
-welcome(); // ok now
+welcome(); // teraz zadziała
 */!*
 ```
 
-Or we could simplify it even further using a question mark operator `?`:
+Możemy też go uprościć za pomocą operatora warunkowego `?`:
 
 ```js run
-let age = prompt("What is your age?", 18);
+let age = prompt("Ile masz lat?", 18);
 
 let welcome = (age < 18) ?
-  function() { alert("Hello!"); } :
-  function() { alert("Greetings!"); };
+  function() { alert("Cześć!"); } :
+  function() { alert("Pozdrowienia!"); };
 
 *!*
-welcome(); // ok now
+welcome(); // działa!
 */!*
 ```
 
 
-```smart header="When to choose Function Declaration versus Function Expression?"
-As a rule of thumb, when we need to declare a function, the first to consider is Function Declaration syntax. It gives more freedom in how to organize our code, because we can call such functions before they are declared.
+```smart header="Kiedy wykorzystywać deklaracje funkcji, a kiedy wyrażenia funkcyjne?"
+Chcąc utworzyć funkcję, zazwyczaj lepiej najpierw rozważyć deklarację funkcji. Daje ona więcej swobody co do organizacji kodu, ponieważ możemy wywołać takie funkcje przed tym jak zostaną zadeklarowane.
 
-That's also better for readability, as it's easier to look up `function f(…) {…}` in the code than `let f = function(…) {…};`. Function Declarations are more "eye-catching".
+Deklaracje funkcji są też czytelniejsze i bardziej "wpadają w oko"; łatwiej znaleźć w kodzie `function f(…) {…}`, niż `let f = function(…) {…};`.
 
-...But if a Function Declaration does not suit us for some reason, or we need a conditional declaration (we've just seen an example), then Function Expression should be used.
+Dopiero jeśli z jakichś powodów deklaracja funkcji nie odpowiada naszym potrzebom (kiedy, jak w przykładzie, potrzebujemy warunkowej deklaracji funkcji), wtedy można rozważyć użycie wyrażenia funkcyjnego.
```

Comment:
Pleace replace 
`Dopiero jeśli z jakichś powodów deklaracja funkcji nie odpowiada naszym potrzebom (kiedy, jak w przykładzie, potrzebujemy warunkowej deklaracji funkcji)` 
by
`Dopiero jeśli z jakichś powodów deklaracja funkcji nie odpowiada naszym potrzebom, albo potrzebujemy warunkowej deklaracji funkcji (jak w przykładzie powyżej)`
 - I think it is more meaningful :)